the-example-app.swift/Services.swift

```diff
@@ -14,11 +15,16 @@ class Services {
 
     public let contentfulStateMachine: StateMachine<ContentfulService>
 
-    public func resetCredentialsToDefault() {
+    public func resetCredentialsAndLocaleToDefault() {
         let defaultCredentials = ContentfulCredentials.default
+
+        // Retain state from last ContentfulService, but ensure we are using a locale that is available in default space.
+        var state = contentful.stateMachine.state
+        let availableLocales = [Contentful.Locale.americanEnglish(), Contentful.Locale.german()]
+        state.locale = availableLocales.contains(contentful.stateMachine.state.locale) ? contentful.stateMachine.state.locale : Contentful.Locale.americanEnglish()
```

Comment:
Can't you just use `state.locale` here?, as you're already extracting `contentful.stateMachine.state` as a variable above.

Reply:
I'm only using `state.locale` if the default space contains that particular locale, so if you are currently viewing Arabic, then switch back to default (so the last `state.locale` = Arabic), the default space doesn't have arabic, so we must switch back to americanEnglish (the default locale). 

The ternary operator is being used in the line of code you commented on:

```
? contentful.stateMachine.state.locale : Contentful.Locale.americanEnglish()
```